src/Text/Pandoc/Readers/Markdown.hs

```diff
@@ -1696,7 +1697,8 @@ subscript = do
     char '~'
     mconcat <$> many1Till (do notFollowedBy spaceChar
                               notFollowedBy newline
-                              inline) (char '~'))
+                              inline) ((char '~') <|> (do guardEnabled Ext_mmd_short_subscript
```

Comment:
This will allow arbitrary inlines in the subscript, e.g. `X~*two and a half*` would be equivalent to the HTML `X<sub><em>two and a half</em></sub>`. I'm not sure how Multimarkdown behaves here. Can you check and add an appropriate test for this case?

Reply:
I had a quick look at the MMD documentation. It states

> The subscript must not contain any whitespace or **punctuation**.

Emphasis mine. I don't think we account for that yet.

Reply:
That makes sense. However, I don't have an easy way to check how Multimarkdown would handle this. As I understand, the reference implementation is a MacOS program (the program costs money, but there is a free preview generator MultiMarkdown QuickLook that should do the job). The only syntax documentation I could find is the cheat sheet that was linked in the original issue, which doesn't address this sort of nesting.
I don't own a Mac device so I can't check it myself. Is this a sort of thing I can ask for help with on pandoc-discuss?

Reply:
Whoops, I'll take a look again. Also, I've now found a Windows version of the tool that's freely available (https://github.com/fletcher/MultiMarkdown-6/releases ) so I can just try it out after all. Sorry!

Reply:
I did some experiments. Including it here for reference:
`x^a.bc` becomes `x<sup>a</sup>.bc`. Same for other non-alphanumeric characters like `!,;-_#@`.
`x^*a*` becomes `x^<em>a</em>`.
`x^a*b*c` becomes `x^<sup>a</sup><em>b</em>c`.

The same thing happens for subscripts. I'll try to implement this accordingly.